=str #17854: fullClose should not Abort

drewhk opened the pull request.

Commit:
=str #17854: fullClose should not Abort

jrudolph commented:
How about splitting it up into two commits, one that fixes the `Abort` problem and one that fixes the `ConfirmedClose` / `Close` sequence problem in `fullClose` mode.

jrudolph commented on akka-stream/src/main/scala/akka/stream/impl/io/TcpConnectionStream.scala:
```diff
@@ -154,17 +142,25 @@ private[akka] abstract class TcpStreamActor(val settings: ActorMaterializerSetti
+          if (tcpInputs.isClosed) {
+            // Reading has stopped, either because of cancel, or PeerClosed, just Close now
+            connection ! Close
+            tryShutdown()
```
These two lines now show up several times. Maybe it should be given a name now... :)

drewhk replied:
well, three times :) I add a name.

drewhk commented:
I am not sure it worth an extra commit, but why not.

Commit:
Fixed wrong test-case:

 - there is no "full-close" if client is still writing, it will always RST (now using lazyEmpty)
 - must Await for bind, otherwise there is a race between connect and bind

ktoso commented:
LGTM, validated on my machine a few hundred times.

drewhk commented:
extra Jenkins run won't hurt
